test_scripts/API/Restructuring_OnResetTimeout/022_BC_DialNumber_without_BC_OnResetTimeout.lua

```diff
@@ -0,0 +1,65 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md
+-- User story: TBD
+-- Use case: TBD
+--
+-- Requirement summary: TBD
+--
+-- Description:
+-- In case:
+-- 1) DialNumber is requested
+-- 2) Some time after receiving DialNumber request on HMI is passed
+-- 3) HMI sends BC.OnResetTimeout(resetPeriod = 15000) to SDL
+-- 4) HMI sends response
+-- SDL does:
+-- 1) SDL must always wait for HMI response to BC.DialNumber as long as it takes and not return GENERIC_ERROR to mobile app
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local paramsForRespFunction = {
+	respTime = 13000,
+	notificationTime = 0,
+	resetPeriod = 12000
+}
+
+local function DialNumber( pRespExpect, pRespTime, pResponseFunc, pParams, pRespParams )
+  if not pRespParams then pRespParams = { success = true, resultCode = "SUCCESS" } end
+  local cid = common.getMobileSession():SendRPC("DialNumber", { number = "#3804567654*" })
+  EXPECT_HMICALL("BasicCommunication.DialNumber", { appID = c.getHMIAppId(), number = "#3804567654*" })
+  :Do(function(_, data)
+    pResponseFunc(data, pParams)
+  end)
+  common.getMobileSession():ExpectResponse(cid, pRespParams)
+  :Timeout(pRespExpect)
+  :ValidIf(function()
+    local respTime = timestamp()
+    local timeBetweenRespAndNot = respTime - c.notificationTime
+    if timeBetweenRespAndNot >= pRespTime - 500 and timeBetweenRespAndNot <= pRespTime + 500 then
+      return true
+    else
+      return false, "Response is received in some unexpected time. Actual time is " .. timeBetweenRespAndNot ..
+      ". Ecpected time is " .. pRespTime
+    end
+  end)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("App registration", common.registerAppWOPTU)
+runner.Step("App activation", common.activateApp)
+runner.Step("Create InteractionChoiceSet", common.createInteractionChoiceSet)
+
+runner.Title("Test")
+runner.Step("Send DialNumber", common.DialNumber,
```

Comment:
@HSavynetska why do you override the _DialNumber_ function in line 31, but here you use _DialNumber_ function from **common** ?

Reply:
@LitvinenkoIra Update